### PR TITLE
Refactor code according to the newest clippy and rustfmt

### PR DESCRIPTION
**Pull request by jnferner**
This PR addresses some issues that clippy and rustfmt had with the code.
As clippy features quite a broad test suite, I wasn't able to fix all warnings.
No commits in this PR change any actual behavior.

### CODE REVIEW

File: src/joint/ball_joint.rs [hunk 1]
```diff
@@ -90,12 +92,12 @@ impl<N: Real> Joint<N> for BallJoint<N> {
     fn integrate(&mut self, params: &IntegrationParameters<N>, vels: &[N]) {
         let angvel = Vector3::from_row_slice(&vels[..3]);
         let disp = UnitQuaternion::new_eps(angvel * params.dt, N::zero());
-        self.rot = disp * self.rot;
+        self.rot *= disp;
```
[the remark this answers is not in the document]
Comment (jnferner) [reply src/joint/ball_joint.rs]: Alright, I will change it. Out of curiosity, would you mind pointing out the code that makes `UnitQuaternion` * `UnitQuaternion` non-commutative? `clippy` is suggesting that the given refactoring is equivalent. If you provide the code, I could submit a bug report.

[the remark this answers is not in the document]
Comment (sebcrozet) [reply src/joint/ball_joint.rs]: Except for few very special cases, all rotation composition are non-comutative.
The code for quaternion multiplication is [there](https://github.com/rustsim/nalgebra/blob/master/src/geometry/quaternion_ops.rs#L160).
I believe clippy just assumes all multiplications for any type are commutative and outputs its warning. It likely does not analyse the multiplication implementation to check if commutativity holds.

[the remark this answers is not in the document]
Comment (jnferner) [reply src/joint/ball_joint.rs]: Thanks for the explanation! I updated the code in question

**Comment by jnferner** [comment]
@sebcrozet I addressed your concerns

**Comment by sebcrozet** [comment]
Great work, thanks!

**Pull request merged**